chalk/core.py

```diff
@@ -189,35 +198,49 @@ def _repr_svg_(self) -> str:
         os.unlink(f.name)
         return svg
 
-    def atop(self, other: "Diagram") -> "Diagram":
+    def _empty_check(
```

Comment:
I don't know if this function will be replaced when moving to envelopes, but I find its logic a bit hard to follow; maybe a short comment would be helpful. Some thoughts:

- I assume that it is needed because `Empty` diagrams do not have a `BoundingBox`, right? It wouldn't work if `Empty` diagrams were to return a fake bounding box (`BoundingBox([Point(0, 0)])`) instead of `None`, would it?
- Are the `fake` bounding boxes ever used? If I understand correctly, if `box1 is None or box2 is None` then the first argument returned is not `None` (it is either `self` or `other`) and the caller function (e.g., `_merge` or `atop`) will return it (while the second and third arguments will be ignored).
- Would it be a bit clearer if the conditions were on diagram instead of the bounding box; that is, `if isinstance(self, Empty)` instead of `if self.get_bounding_box() is None`?

Reply:
Yeah this is a little awkward. It is based on the Yorgey paper that defines envelopes as Options 

![image](https://user-images.githubusercontent.com/35882/176225950-85a7a61d-2914-4108-914c-1dbf4f721b63.png)

Why don't I push this into the Envelope class and its mappend definition. I will fix it in the next PR?